scripts/create_test_subset.py

```diff
@@ -116,204 +232,319 @@ def main(
+    original_project_subset_data = query(
+        QUERY_ALL_DATA, {'project': project, 'sids': final_subset_sids}
     )
 
+    # Pull Participant Data
+    participant_data = []
+    participant_ids: list = []
+    for sg in original_project_subset_data.get('project').get('samples'):
+        participant = sg.get('participant')
+        if participant:
+            participant_data.append(participant)
+            participant_ids.append(participant.get('externalId'))
+
     # Populating test project
     target_project = project + '-test'
-    logger.info('Checking any existing test samples in the target test project')
-
-    test_sample_by_external_id = _process_existing_test_samples(target_project, samples)
-
-    try:
-        seq_infos: list[dict[str, str]] = assayapi.get_assays_by_criteria(
-            body_get_assays_by_criteria=BodyGetAssaysByCriteria(
-                sample_ids=sample_ids,
-            )
-        )
-    except exceptions.ApiException:
-        seq_info_by_s_id = {}
-    else:
-        seq_info_by_s_id = {seq['sample_id']: seq for seq in seq_infos}
-
-    analysis_by_sid_by_type: dict[str, dict[str, dict[str, str]]] = {
-        'cram': {},
-        'gvcf': {},
-    }
-    for a_type, analysis_by_sid in analysis_by_sid_by_type.items():
-        try:
-            # TODO: fix this
-            analyses: list[
-                dict[str, str]
-            ] = aapi.get_latest_analysis_for_samples_and_type(
-                project=project,
-                analysis_type=str(a_type),
-                request_body=sample_ids,
-            )
-        except exceptions.ApiException:
-            traceback.print_exc()
-        else:
-            for a in analyses:
-                analysis_by_sid[a['sample_ids'][0]] = a
-        logger.info(f'Will copy {a_type} analysis entries: {analysis_by_sid}')
 
     # Parse Families & Participants
-    participant_ids = [int(sample['participant_id']) for sample in samples]
     if skip_ped:
         # If no family data is available, only the participants should be transferred.
-        external_participant_ids = transfer_participants(
-            initial_project=project,
+        upserted_participant_map = transfer_participants(
             target_project=target_project,
-            participant_ids=participant_ids,
+            participant_data=participant_data,
         )
 
     else:
         family_ids = transfer_families(project, target_project, participant_ids)
-        external_participant_ids = transfer_ped(project, target_project, family_ids)
+        upserted_participant_map = transfer_ped(project, target_project, family_ids)
 
-    external_sample_internal_participant_map = get_map_ipid_esid(
-        project, target_project, external_participant_ids
+    existing_data = query(EXISTING_DATA_QUERY, {'project': target_project})
+
+    samples = original_project_subset_data.get('project').get('samples')
+    transfer_samples_sgs_assays(
```

Comment:
Is there a way to add a `dry_run` option so that these `transfer_` methods print what they will do, rather than actually doing the thing? For example rather than calling `aapi.create_analaysis` it would log the details of the analysis that will be created.